packages/matter.js/src/protocol/MessageExchange.ts

```diff
@@ -202,18 +228,42 @@ export class MessageExchange<ContextT> {
             );
         }
         if (requiresAck) {
+            // We still have a message to ack, so ack this one as standalone ack directly
+            if (this.receivedMessageToAck !== undefined) {
+                this.receivedMessageAckTimer.stop();
+                await this.sendStandaloneAckForMessage(this.receivedMessageToAck);
+                return;
+            }
             this.receivedMessageToAck = message;
+            this.receivedMessageAckTimer.start();
         }
         await this.messagesQueue.write(message);
     }
 
     async send(messageType: number, payload: ByteArray, options?: ExchangeSendOptions) {
-        const { expectAckOnly = false, minimumResponseTimeoutMs, requiresAck } = options ?? {};
-        if (this.sentMessageToAck !== undefined)
+        const {
+            expectAckOnly = false,
+            minimumResponseTimeoutMs,
+            requiresAck,
+            includeAcknowledgeMessageId,
+        } = options ?? {};
+        if (messageType === MessageType.StandaloneAck && requiresAck) {
+            throw new MatterFlowError("A standalone ack must not send reliable.");
```

Comment:
This message is a little hard to parse, maybe "A standalone ack may not require acknowledgement"?

Reply:
done